scripts/create-darwin-volume.sh

```diff
@@ -83,6 +61,64 @@ suggest_report_error(){
     echo "       please report this @ https://github.com/nixos/nix/issues" >&2
 }
 
+generate_mount_command(){
+    if test_filevault_in_use; then
+        printf "    <string>%s</string>\n" /bin/sh -c '/usr/bin/security find-generic-password -a "Nix Volume" -w | /usr/sbin/diskutil apfs unlockVolume "Nix Volume" -mountpoint /nix -stdinpassphrase'
+    else
+        printf "    <string>%s</string>\n" /usr/sbin/diskutil mount -mountPoint /nix "Nix Volume"
+    fi
+}
+
+generate_mount_daemon(){
+    cat << EOF
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+  <key>RunAtLoad</key>
+  <true/>
+  <key>KeepAlive</key>
+  <dict>
+    <key>PathState</key>
+    <dict>
+      <key>/nix/var/nix</key>
+      <false/>
+    </dict>
+  </dict>
+  <key>Label</key>
+  <string>org.nixos.darwin-store</string>
+  <key>ProgramArguments</key>
+  <array>
+$(generate_mount_command)
+  </array>
+</dict>
+</plist>
+EOF
+}
+
+prepare_darwin_volume_password(){
+    sudo /usr/bin/expect -f - "$1" "$2" << 'EOF'
+log_user 0
+set VOLUME [lindex $argv 0];
+set UUID [lindex $argv 1];
+set PASSPHRASE [exec /usr/bin/ruby -rsecurerandom -e "puts SecureRandom.hex(32)"]
+
+# Cargo culting: people recommend this; not sure how necessary
+set send_slow {1 .1}
+spawn /usr/bin/sudo /usr/bin/security add-generic-password -a "$VOLUME" -s "$UUID" -D "$VOLUME encryption password" -U -w
+expect {
+    "password data for new item: " {
+        send -s -- "$PASSPHRASE\r"
+        expect "retype password for new item: " {
+            send -s -- "$PASSPHRASE\r"
+        }
+    }
+}
```

Comment:
I assume we're using `expect` here purely to avoid having the password show up in the command-line arguments in case anyone is sniffing it?

I don't think we need to worry about that. The volume is going to be mounted the entire time the computer is running, so any process that can sniff the command-line arguments can also just look at the volume while it's mounted anyway. We should ditch the complexity of `expect` (and the risk that it might be deprecated too since `tclsh` is) and just provide the passphrase on the command-line.

Reply:
I'll treat the "expect" comments here as contingent on the answer to this one, since they're moot if we throw it overboard.

1. I roughly agree with respect to my own system. 
2. My animating _concern_ is that passing the password will ward off prospective Nixers with policy/compliance/superstitious limitations.
3. I admit my own concern is basically superstitious, so I'm eager to defer to any consensus among senior Nix community members.

Reply:
It occurs to me that since `/usr/bin/security` has access to the password, keeping it out of the command-line does literally nothing at all because any process that cared could just immediately invoke `/usr/bin/security` to read the password.

Reply:
> I assume we're using `expect` here purely to avoid having the password show up in the command-line arguments in case anyone is sniffing it?
> 
> I don't think we need to worry about that. The volume is going to be mounted the entire time the computer is running, so any process that can sniff the command-line arguments can also just look at the volume while it's mounted anyway. We should ditch the complexity of `expect` (and the risk that it might be deprecated too since `tclsh` is) and just provide the passphrase on the command-line.

I finally found a way to have and eat our cake, here (dump the expect dependency, without the password ending up in the process table). Apparently `security -i` will happily run commands from a heredoc. See b38e85f 

> It occurs to me that since `/usr/bin/security` has access to the password, keeping it out of the command-line does literally nothing at all because any process that cared could just immediately invoke `/usr/bin/security` to read the password.

The process still needs root to read the system keychain, but this is generally still a point of concern. The daemon component probably needs to get replaced with a binary at some point.

I've tried a few things here:
1. Find some imperative command that uses whatever auth mechanism is used when entries from fstab are mounted. 
    - ~`mount`~
    - ~`diskutil mount`~ 
    - ~`diskutil apfs unlockVolume` with no passphrase~
    - `automount`? Jury is still out, but I'm not optimistic.
2. Scripts can be added to the exemption dialog, but aren't actually allowed access.